Build fails on OpenBSD 7.2

This seems potentially related to: https://github.com/gbdev/rgbds/issues/1091

This will fix the problem, but I think there may be a better way, and someone more familiar with the codebase may have a better solution:
```
diff --git a/Makefile b/Makefile
index d39906ba..18f01279 100644
--- a/Makefile
+++ b/Makefile
@@ -42,7 +42,7 @@ CXXFLAGS      ?= -O3 -flto -DNDEBUG
 REALCFLAGS     := ${CFLAGS} ${WARNFLAGS} -std=gnu11 -I include \
                   -D_POSIX_C_SOURCE=200809L -D_ISOC11_SOURCE
 REALCXXFLAGS   := ${CXXFLAGS} ${WARNFLAGS} -std=c++17 -I include \
-                  -D_POSIX_C_SOURCE=200809L -fno-exceptions -fno-rtti
+                  -fno-exceptions -fno-rtti
 # Overridable LDFLAGS
 LDFLAGS                ?=
 # Non-overridable LDFLAGS
```

Full build output:
```
DEFS=-Dlr.type=ielr -Dparse.lac=full -Dparse.error=verbose -Dparse.error=detailed -Dapi.token.raw=true
src/link/object.c:672:14: warning: result of comparison of constant 1152921504606846975 with expression of type 'unsigned int' is always false [-Wtautological-constant-out-of-range-compare]
        if (nbFiles > SIZE_MAX / sizeof(*nodes))
            ~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
In file included from src/gfx/main.cpp:16:
In file included from /usr/include/c++/v1/fstream:188:
In file included from /usr/include/c++/v1/istream:163:
In file included from /usr/include/c++/v1/ostream:140:
In file included from /usr/include/c++/v1/locale:218:
/usr/include/c++/v1/__bsd_locale_fallbacks.h:122:17: error: use of undeclared identifier 'vasprintf'; did you mean 'vsprintf'?
    int __res = vasprintf(__s, __format, __va);
                ^
/usr/include/c++/v1/cstdio:124:9: note: 'vsprintf' declared here
using ::vsprintf _LIBCPP_USING_IF_EXISTS;
        ^
In file included from src/gfx/main.cpp:16:
In file included from /usr/include/c++/v1/fstream:188:
In file included from /usr/include/c++/v1/istream:163:
In file included from /usr/include/c++/v1/ostream:140:
In file included from /usr/include/c++/v1/locale:218:
/usr/include/c++/v1/__bsd_locale_fallbacks.h:122:27: error: cannot initialize a parameter of type 'char *' with an lvalue of type 'char **'
    int __res = vasprintf(__s, __format, __va);
                          ^~~
/usr/include/stdio.h:269:21: note: passing argument to parameter here
int      vsprintf(char *, const char *, __va_list);
                        ^
src/gfx/main.cpp:351:31: warning: result of comparison of constant 256 with expression of type 'std::array<unsigned char, 2>::value_type' (aka 'unsigned char') is always false [-Wtautological-constant-out-of-range-compare]
                        if (options.baseTileIDs[0] >= 256) {
                            ~~~~~~~~~~~~~~~~~~~~~~ ^  ~~~
src/gfx/main.cpp:367:31: warning: result of comparison of constant 256 with expression of type 'std::array<unsigned char, 2>::value_type' (aka 'unsigned char') is always false [-Wtautological-constant-out-of-range-compare]
                        if (options.baseTileIDs[1] >= 256) {
                            ~~~~~~~~~~~~~~~~~~~~~~ ^  ~~~
src/gfx/main.cpp:482:27: warning: result of comparison of constant 256 with expression of type 'uint8_t' (aka 'unsigned char') is always false [-Wtautological-constant-out-of-range-compare]
                        if (options.nbPalettes > 256) {
                            ~~~~~~~~~~~~~~~~~~ ^ ~~~
3 warnings and 2 errors generated.
*** Error 1 in /home/rupert/src/rgbds (Makefile:169 'src/gfx/main.o': @c++ -O2 -pipe  -Wall -pedantic -std=c++17 -I include  -D_POSIX_C_SOURCE...)
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -41,7 +41,6 @@ else()
     # does not recognize this yet.
     add_compile_options(-Wno-gnu-zero-variadic-macro-arguments)
   endif()
-  add_definitions(-D_POSIX_C_SOURCE=200809L)
   if(SANITIZERS)
     set(SAN_FLAGS -fsanitize=shift -fsanitize=integer-divide-by-zero
                   -fsanitize=unreachable -fsanitize=vla-bound

--- a/Makefile
+++ b/Makefile
@@ -26,19 +26,16 @@ PNGLDLIBS	:= `${PKG_CONFIG} --libs-only-l libpng`
 # Note: if this comes up empty, `version.cpp` will automatically fall back to last release number
 VERSION_STRING	:= `git --git-dir=.git describe --tags --dirty --always 2>/dev/null`
 
-WARNFLAGS	:= -Wall -pedantic -Wno-unknown-warning-option \
-                   -Wno-gnu-zero-variadic-macro-arguments
+WARNFLAGS	:= -Wall -pedantic -Wno-unknown-warning-option -Wno-gnu-zero-variadic-macro-arguments
 
 # Overridable CXXFLAGS
 CXXFLAGS	?= -O3 -flto -DNDEBUG
 # Non-overridable CXXFLAGS
-REALCXXFLAGS	:= ${CXXFLAGS} ${WARNFLAGS} -std=c++2a -I include \
-		   -D_POSIX_C_SOURCE=200809L -fno-exceptions -fno-rtti
+REALCXXFLAGS	:= ${CXXFLAGS} ${WARNFLAGS} -std=c++2a -I include -fno-exceptions -fno-rtti
 # Overridable LDFLAGS
 LDFLAGS		?=
 # Non-overridable LDFLAGS
-REALLDFLAGS	:= ${LDFLAGS} ${WARNFLAGS} \
-		   -DBUILD_VERSION_STRING=\"${VERSION_STRING}\"
+REALLDFLAGS	:= ${LDFLAGS} ${WARNFLAGS} -DBUILD_VERSION_STRING=\"${VERSION_STRING}\"
 
 # Wrapper around bison that passes flags depending on what the version supports
 BISON		:= src/bison.sh

--- a/include/platform.hpp
+++ b/include/platform.hpp
@@ -56,4 +56,9 @@
 	#define setmode(fd, mode) (0)
 #endif
 
+// MingGW and Cygwin need POSIX functions which are not standard C explicitly enabled
+#if defined(__MINGW32__) || defined(__MINGW32__) || defined(__CYGWIN__)
+	#define _POSIX_C_SOURCE 200809L
+#endif
+
 #endif // RGBDS_PLATFORM_HPP

